### PR TITLE
Add user profile and dat database

### PR DESCRIPTION
**Pull request by karissa**
This adds a simple user profile database api.

### CODE REVIEW

**Comment by karissa** [comment]
right now this requires `township-token@2.2.0`, so until it gets published you have to use the github version

**Comment by karissa** [comment]
closes #321 #325 #324

File: server/auth/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      jsonBody(req, res, function (err, body) {
+        if (err) return error(401, err.message).pipe(res)
+        if (!body.email) return error(400, 'must specify email').pipe(res)
+        verify(body.email, function (err) {
+          if (err) return error(401, err.message).pipe(res)
+          ship.register(req, res, {body: body}, function (err, statusCode, obj) {
+            if (err) return error(400, err.message).pipe(res)
+            db.models.users.create({email: body.email, id: obj.key}, function (err, body) {
+              if (err) return error(400, err.message).pipe(res)
+              return send(body).pipe(res)
```
Comment (joehand) [inline server/auth/index.js]: We need to send `obj` back via response so users can get their tokens.

Comment (karissa) [inline server/auth/index.js]: good catch!

**Pull request merged**